### PR TITLE
Try alternative process name if cluster could not be formed

### LINKED ISSUE #11038: Cannot create quorum queues on vhosts due to `vhost_queue` process name
### Describe the bug

The internal QQ process names are of the form `vhost_queue`. This could cause issues when vhost and queue names contain underscores.

### Reproduction steps

1. Create 2 vhosts called "foo" and "foo_bar"
2. On vhost "foo", create a QQ called "bar_baz"
3. On vhost "foo_bar", create a QQ called "baz"
4. Observe the following error:
image

### Expected behavior

User can successfully create both QQs.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexnkdev**
## Proposed Changes

Retry cluster creation with alternative name, if replica process name conflicts between queues in different virtual hosts.

Fixes #11038 

## Types of Changes

What types of changes does your code introduce to this project?

- [x] Bug fix (non-breaking change which fixes issue #11038)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

**Comment by michaelklishin** [comment]
The Actions failures have to do with secret access for external contributions. We will run the tests as part of the QA process, don't sweat it.

**Comment by michaelklishin** [comment]
@alexnkdev can you please rebase this PR on top of `main`? Then I will submit a PR so that the suites do run. Your commits will still be credit to you, of course.

**Comment by alexnkdev** [comment]
@michaelklishin done

File: deps/rabbit/src/rabbit_quorum_queue.erl [hunk 1]
```diff
@@ -230,27 +230,48 @@ declare(Q, _Node) when ?amqqueue_is_quorum(Q) ->
...
-        rabbit_queue_location:select_leader_and_followers(Q, QuorumSize),
+    rabbit_log:debug("Will start up to ~w replicas for quorum ~ts with leader on node '~ts'",
+                     [QuorumSize, rabbit_misc:rs(QName), LeaderNode]),
+    case internal_declare(Q, RaName, LeaderNode, FollowerNodes) of
+        {error, cluster_not_formed, Queue} ->
+            _ = rabbit_amqqueue:internal_delete(Queue, ActingUser),
+            AlternativeRaName = rabbit_data_coercion:to_atom(ra:new_uid(rabbit_queue_type_util:name_concat(QName))),
+            rabbit_log:debug("Could not form cluster ~ts, trying alternative name ~ts",
+                     [RaName, AlternativeRaName]),
+            case internal_declare(Q, AlternativeRaName, LeaderNode, FollowerNodes) of
```
Comment (michaelklishin) [inline deps/rabbit/src/rabbit_quorum_queue.erl:253]: This doesn't handle every error `internal_declare/4` can return.

**Review by michaelklishin — CHANGES_REQUESTED** [review]
There is an xref failure:

```
Undefined function rabbit_queue_type_util:name_concat/1 called by rabbit_quorum_queue:start_cluster/1
```

And indeed, `name_concat/1` [is not exported](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_queue_type_util.erl).

**Comment by michaelklishin** [comment]
To run xref and Dialyzer, use

``` shell
bazel test //deps/rabbit:xref
bazel test //deps/rabbit:dialyze
```

with Bazel or

``` shell
gmake xref
# gmake dialyze is broken with Make IIRC
```

File: deps/rabbit/src/rabbit_quorum_queue.erl (hunk 1 above)
Comment (kjnilsson) [reply deps/rabbit/src/rabbit_quorum_queue.erl:253]: it also doesn't handle the partial case where there is a name conflict in a minority subset of the members.

**Comment by michaelklishin** [comment]
This cannot be merged due to an `xref` failure (a function that does not exist is called) and https://github.com/rabbitmq/rabbitmq-server/pull/11866#discussion_r1705704486.

Closing after a few weeks without any activity.

**Pull request closed without merging**